### ISSUE TITLE
how to use tispark write to tidb?

### ISSUE THREAD

**Issue by dalinqin** [pingcap/tispark#205]
I'm new to tidb ,the tispark example works for me , I can use  following code to query data from tidb :
```
scala> import org.apache.spark.sql.TiContext
import org.apache.spark.sql.TiContext

scala> val ti = new TiContext(spark)
ti: org.apache.spark.sql.TiContext = org.apache.spark.sql.TiContext@702ab027

scala> ti.tidbMapDatabase ("TPCH_001")

scala> spark.sql("select count(*) from lineitem").show()
+--------+
|count(1)|
+--------+
|   60175|
+--------+

```
my question is :if I have a dataframe existing ,how to write the dataframe into tidb?

**Comment by zhexuany** [comment]
Hello there, 

Just some clarification I want to made. `TiSpark` can only query data from `TiKV` with the help from `PD`. When you are running `TiSpark` service, it does not require `TiDB` being present. 

If you are asking about can `TiSpark` write data into `TiKV`, the answer is no. We may support this in future, but we do not have enough manpower to do this. Fortunately, you can use `JDBC` for writing data into `TIDB`. 

I hope above can help you. Thanks for you interesting in using `TiSpark` project.

**Comment by dalinqin** [comment]
thank you for the explain, using spark to write through jdbc is kinda slow ,I just hope someday you guys can let user using spark write dataframe directly to tidb.

**Comment by ilovesoup** [comment]
We have an ongoing project write batch directly to TiKV's underlying file and that will speed up table dump. We will ping you once it's done (2018 Q1).

**Comment by dalinqin** [comment]
ok,that's really great ,currently I'm using spark jdbc api writing to tikv ,which really suffers a lot ,I also checked tidb tools such as myloader which not working for csv format ,so basicly if you original data is not in mysql ,there is no fast(parallel)  way to mass batch insert into tikv right now ,that might make many users hesitate to use this great new sql db.

**Comment by Novemser** [comment]
@dalinqin CSV format file also made me crazy once upon a time, to solve this problem, I wrote a tiny program to convert CSV format file to `mydumper` format [here](https://github.com/Novemser/tidb-tpch-dataloader). After this conversion, loading data into TiKV cluster via `loader` becomes fast and easy. 

If you really need to load CSV file to your TiKV right now, I recommend you check out my [solution](https://github.com/Novemser/tidb-tpch-dataloader)(though may be buggy and shabby..) and see whether it can help in your scenario(e.g. Save your dataframe and schema to CSV format and load them), of course some minor modification on the script file is needed.

Thanks for your attention on TiDB and TiSpark.

Good lock. : )

**Comment by shenli** [comment]
@Novemser The improve the performance of `Load Data` command by 10 times. But still slower than `loader`.
You can try the lastest master.

**Comment by Novemser** [comment]
@shenli Thanks, will try the next time.

**Comment by dalinqin** [comment]
@Novemser  that's really sweet ,I have done the most time consuming job though .what I did is install a single mysql server and load the csv into that database ,then using mydumper(lighting fast for the single instance mysql5.7) and loaded the dump file  into the tikv database , have to say load itself is pretty slow either.

I use the command  
`./loader -u root -h 127.0.0.1 -P 4000 -t 16  -B sparkdb -d /udata/mysql_dmp/ ` to load the data
( I was trying to use -q 200000 which is not supported)

during the upload ,I got endless message below ,it take several hours to upload 0.3 billion rows.
```
2018/01/20 13:40:07 db.go:95: [warning] transaction execution costs 1.958618 seconds
2018/01/20 13:40:09 db.go:95: [warning] transaction execution costs 1.829700 seconds
2018/01/20 13:40:11 db.go:95: [warning] transaction execution costs 1.735041 seconds
2018/01/20 13:40:13 db.go:95: [warning] transaction execution costs 1.837624 seconds
2018/01/20 13:40:14 db.go:95: [warning] transaction execution costs 1.762402 seconds

```

can someone let me why the load is so slow?  `show processlist ` showing only one process is busy which means -t 16 is not working .
